src/wp-admin/css/customize-controls.css

```
@@ -362,11 +362,16 @@ body.trashing #publish-settings {
 .date-time-fields .date-input.day,
 .date-time-fields .date-input.hour,
 .date-time-fields .date-input.minute {
-	width: 46px;
+	width: 52px;
+}
+
+.wp-core-ui .date-time-fields .day-fields.clear select,
+.wp-core-ui .date-time-fields .time-fields.clear select {
```

```suggestion
.customize-control-date_time select {
```

This only needs a class and an element to override `.wp-core-ui select { vertical-align: middle; }` in `forms.css`. The `.customize-control-date_time` class would cover both the date and the time.